content/learn/book/storing-data/queries.md

```diff
@@ -6,10 +6,260 @@ weight = 4
+    special_entity.0 = special_entity_id;
+}
+
+fn print_selected_entity_name(query: Query<&Name>, special_entity: Res<SelectedEntity>){
+    if let Ok(name) = query.get(special_entity.0){
+        info!("{name} is selected.");
+    } else {
+        warn!("Selected entity {} has been despawned, or does not have a Name component", special_entity.0);
+    }
+}
+```
+
+As the example shows, you can store the retrieved `Entity` identifiers inside of components or resources,
+creating flexible connections between entities and lookup tables.
+Inside of Bevy itself, this pattern is combined with [hooks] to make it more robust
+and exposed to users as [relations].
+
+The other common way to get an [`Entity`] is to take advantage of its [`QueryData`] implementation,
+allowing you to determine the identifier for entities in queries that you are iterating over.
+
+```rust
+# use bevy::prelude::*;
+#
+#[derive(Component)]
+struct Enemy;
+
+// Note: no `&` for Entity as a QueryData!
+fn despawn_all_enemies(enemies: Query<Entity, With<Enemy>>, mut commands: Commands){
+    for enemy_entity in enemies.iter(){
+        commands.entity(enemy_entity).despawn();
+    }
+}
+```
+
+[hooks]: ../control-flow/hooks.md
+[relations]: ./relations.md
+
+## Working with singleton entities
+
+From time-to-time, you may find yourself writing systems that expect there to be only a single matching entity.
+This might be a player, the sun, or something more abstract like your camera.
+
+While you could iterate over a query of length one, this can be confusing to read and feel a bit silly.
+To make working with these patterns more comfortable, Bevy provides two tools:
+`Query::single` and the `Single` system param.
+Let's try writing the same simple system in each of the three ways.
+
+```rust
+# use bevy::prelude::*;
+#
+#[derive(Component)]
+struct Life(u32);
+
+fn kill_player_when_dead_query_iter(player_query: Query<(Entity, &Life), With<Player>>, mut commands: Commands) {
+    for (player_entity, player_life) in player_query.iter(){
+        if player_life.0 == 0 {
+            commands.entity(player_entity).despawn();
+        }
+    }
+}
+
+fn kill_player_when_dead_query_single(player_query: Query<(Entity, &Life), With<Player>>, mut commands: Commands) {
+    let Ok((player_entity, player_life)) = player.single() else {
+        // We could instead use the ? operator and return an error;
+        // see the error handling chapter        
+        return;
+    }
+
+    if player_life.0 == 0 {
+        commands.entity(player_entity).despawn();
+    }
+}
+
+// This system will be skipped unless there is exactly one matching player entity
+// so there's no need to handle the error case in the system
+fn kill_player_when_dead_query_single(player: Single<(Entity, &Life), With<Player>>, mut commands: Commands) {
+    // We have to dereference out of the Single smart pointer
+    // before we can use destructuring assignment
```

Comment:
```suggestion
    // before we can use destructuring assignment to access the individual components
```
I think this makes it clear even to people who do not recognise what part of that line is a "destructuring assignment".